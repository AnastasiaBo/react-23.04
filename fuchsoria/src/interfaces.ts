export interface IMessage {
  id?: string;
  author: string;
  text: string;
  authorAccess?: string;
  isBot?: boolean;
}

export interface IChatListItem {
  id: number | string;
  title: string;
  description: string;
}

export interface IChatListProps {
  items: IChatListItem[];
}

export interface IChatContainerProps {
  noContent?: boolean;
  history?: unknown;
  location?: unknown;
  match?: {
    isExact: boolean;
    params: {
      chatId: string;
    };
    path: string;
    url: string;
  };
}

export interface IChatContainerState {
  chats: IChats;
  chatList?: IChatListItem[];
}

export interface IMessageListProps {
  messages: IMessage[];
}

export interface IMessageProps extends IMessage {}

export interface IChatFormProps {
  handleSubmit: (author: string, text: string) => void;
}

export interface IChat {
  title: string;
  messages: IMessage[];
}

export interface IChats {
  [key: string]: IChat;
}
