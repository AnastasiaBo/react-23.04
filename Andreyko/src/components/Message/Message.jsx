import React from 'react';
import './style.css'

class Message extends React.Component {
  constructor() {
    super();

    this.state = {
      messages: [],
    };
  }

  addMessage() {
    const { messages } = this.state;

    this.setState({
      messages: [...messages, 'hello']
    })
  }

  render() {
    const { messages } = this.state;
    return (
      <div>
        <button onClick={() => this.addMessage()} className='btn-send'>Отправить запрос</button>
        <div>
          {messages.map((item, i) => (
            <p key={i} className='message'>
              {item + ' ' + i}
            </p>
          ))}
        </div>
      </div>
    )
  }
}

export default Message;