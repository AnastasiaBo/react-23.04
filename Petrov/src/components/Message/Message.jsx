import React from "react";
import PropTypes from 'prop-types';
import { Container } from '@material-ui/core';
import styles from './index.css';
import MessageItem from '../MessageItem';
class Message extends React.PureComponent {


  
  render() {

  return (
    <Container maxWidth="sm" component="ul" display="flex" className={styles.list} style={{display: 'flex'}}>
    
      {this.props.words.map(({ id, message, author }, index) =>
        <MessageItem key={id} text={message} author={author} />
      )}
     
    </Container>
  );

  }
}
Message.propTypes = {
  id: PropTypes.number,
  message: PropTypes.string,
  author: PropTypes.string,
};
export default Message;
