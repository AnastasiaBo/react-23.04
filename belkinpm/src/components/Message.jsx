import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Message extends Component {
  static propTypes = {
    text: PropTypes.string.isRequired,
  };

  render() {
    return <div>{this.props.text}</div>;
  }
}

export default Message;
