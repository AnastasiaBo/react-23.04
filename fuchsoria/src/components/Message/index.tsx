import React from 'react';
import cn from 'classnames';
import { IMessageProps } from '../../interfaces';
import styles from './styles.module.scss';

export default function Message({ author, text, isSelf }: IMessageProps) {
  return (
    <div className={cn([styles.message, { [styles.messageSelf]: isSelf }])}>
      <div className={styles.messageAuthor}>{author}</div>
      {text}
    </div>
  );
}
