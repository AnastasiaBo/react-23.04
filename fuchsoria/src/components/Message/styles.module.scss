.message {
  padding: 10px 20px;
  margin-bottom: 10px;
  text-align: right;
  background-color: #ffff;
  border-radius: 10px;
  max-width: 400px;

  &:last-child {
    margin-bottom: 0;
  }

  &_bot {
    text-align: left;
    align-self: flex-start;
  }

  &__author {
    text-transform: capitalize;
    font-weight: 700;
  }
}
