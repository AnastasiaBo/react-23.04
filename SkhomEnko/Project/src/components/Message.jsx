import React from 'react'
import PropTypes from "prop-types"
import moment from 'moment'
import './Message.css'

function Message(props) {
    const {
      id,
      author,
      message,
      showTimestamp,
      timestamp,
      isMine
    } = props.data

    const fromTimestamp = moment(timestamp).fromNow()
    const formatedTimestamp = moment(timestamp).format('llll')


    return (   
      <div className={`message${isMine ? ' mine' : ''}`} key={id}>
        {
          showTimestamp &&
            <div className="timestamp">
              { formatedTimestamp }
            </div>
        }

        <div className="bubble-container">
          <div className="bubble" title={formatedTimestamp}>
            <b>{ author }</b> <i>{ fromTimestamp }</i> <br />
            { message }
          </div>
        </div>
      </div>
    )
}

Message.propTypes = {
  data: PropTypes.object.isRequired
}

export default React.memo(Message)