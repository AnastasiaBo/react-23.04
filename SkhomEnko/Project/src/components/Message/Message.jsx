import React, {useState} from 'react'
import PropTypes from "prop-types"
import dayjs from 'dayjs'
import './Message.css'
import clsx from 'clsx'
import useInterval from '../hooks/useInterval'

function Message(props) {
    const {
      author,
      message,
      showTimestamp,
      timestamp,
      isMine
    } = props.data
    
    const [fromTimestamp, setFromTimestamp] = useState(dayjs(timestamp).fromNow())
    const formatedTimestamp = dayjs(timestamp).format('llll')

    useInterval(() => {
      if (fromTimestamp !== dayjs(timestamp).fromNow()) setFromTimestamp(dayjs(timestamp).fromNow())
    }, 2000)
   
    return (   
      <div className={clsx('message', {'mine': isMine})}>
        {
          showTimestamp &&
            <div className="timestamp">
              { formatedTimestamp }
            </div>
        }

        <div className="bubble-container">
          <div className="bubble" title={formatedTimestamp}>
            <b>{ author }</b> <i>{ fromTimestamp }</i> <br />
            { message }
          </div>
        </div>
      </div>
    )
}

Message.propTypes = {
  data: PropTypes.shape({
    // id прилетает, но в самом (данном) компоненте он не задействован, используется только снаружи для итерирования
    // id: PropTypes.number,
    author: PropTypes.string,
    message: PropTypes.string,
    showTimestamp: PropTypes.bool,
    timestamp: PropTypes.number,
    isMine: PropTypes.bool
  }).isRequired
}

export default React.memo(Message)