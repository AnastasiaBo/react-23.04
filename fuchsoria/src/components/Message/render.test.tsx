import React from 'react';
import { render } from '@testing-library/react';
import { IMessage } from '../../interfaces';
import Message from '../Message';

const mockData: IMessage = {
  id: 'w7i33e48tr834',
  author: 'Jest',
  text: 'Jest example message',
  authorAccess: 'user',
};

describe('<Message />', () => {
  it('Component should render', async () => {
    const { container } = render(
      <Message
        key={mockData.id}
        author={mockData.author}
        text={mockData.text}
        isBot={mockData.authorAccess === 'bot'}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
