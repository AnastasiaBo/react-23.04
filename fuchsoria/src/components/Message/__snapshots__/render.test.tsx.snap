// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Message /> Component should render 1`] = `
<div>
  <div
    class="message"
  >
    <div
      class="messageAuthor"
    >
      Jest
    </div>
    Jest example message
  </div>
</div>
`;
